software/base/src/main/java/org/apache/brooklyn/entity/software/base/SoftwareProcess.java

```diff
@@ -241,6 +246,7 @@
     @SetFromFlag("runtimeTemplates")
     MapConfigKey<String> RUNTIME_TEMPLATES = new MapConfigKey.Builder<String>(String.class, "templates.runtime")
             .description("Mapping of templates, to be filled in and copied before customisation, to destination name relative to runDir") 
+            .parentInheritance(ConfigInheritance.NONE)
```

Comment:
Not sure about the file uploads. I suggest to leave them as is. The worst that can happen is to end up with some unused files on the machine. I'd even go as far as changing to merge behaviour :).
+1 for the rest.